public/app/features/variables/datasource/actions.ts

```diff
@@ -18,10 +18,8 @@ export const updateDataSourceVariableOptions = (
   identifier: VariableIdentifier,
   dependencies: DataSourceVariableActionDependencies = { getDatasourceSrv: getDatasourceSrv }
 ): ThunkResult<void> => async (dispatch, getState) => {
-  const sources = dependencies
-    .getDatasourceSrv()
-    .getList({ metrics: true, variables: false })
-    .map(toDataSourceSelectItem);
+  const list = dependencies.getDatasourceSrv().getList({ metrics: true, variables: false });
+  const sources = list.map(toDataSourceSelectItem);
```

Comment:
I do not think there is any value in mapping this to DataSourceSelectItem (it's sort of a deprecated type, now that DataSourcePicker handles this internally)

Reply:
Yeah, good suggestion. I'll fix it.

Reply:
Fixed!